### ISSUE TITLE
Add `overviewPath` to asset docs schema

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-platform#1015]
**Summary**

Design spec: https://www.figma.com/file/Cs94zXOUUEizXNdDz263OA/Component-demo?node-id=114%3A169975

Because the Storybook embed will be achieved through a new MDX component, we need a file to use that new component.

We can add an `overviewPath` to `asset.docs` in the schema to render content immediately below the dashboard. See: https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#asset-docs

**Acceptance criteria**

What requirements must be met to complete this request?

- [x] Add `overviewPath`
- [x] Render that markdown content immediately below the dashboard in the asset overview page template

(We might need to create a new `Contributing > Assets` page to house that and other documentation templates.)

**Comment by mattrosno** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Development-624c3fc8b7972e001130fef7/issues/carbon-design-system/carbon-platform/1015?planning-poker) with ZenHub @alisonjoseph @francinelucca @andreancardona

**Comment by francinelucca** [comment]
@kingtraceyj @aubrey-oneal what should happen here when the specified overview content path doesn't exist? (404)? 🤔 since this isn't the entire page, just a section

**Comment by alisonjoseph** [comment]
I think if it doesn't exist then we just don't display content, there might be some components that don't have this file, at least to start with, and that's ok.

**Comment by francinelucca** [comment]
I agree we can just not display anything, wanted to confirm with design if that's ok?

**Comment by mattrosno** [comment]
The content of this file get inserted in the slot in the Overview page below the dashboard. There will always be an Overview page tab and dashboard. So if there's no `overviewPath` indexed, it's a no-op.

**Comment by francinelucca** [comment]
> The content of this file gets inserted in the slot in the Overview page below the dashboard. There will always be an Overview page tab and dashboard. So if there's no `overviewPath` indexed, it's a no-op.

yeah so I'm asking what if there is an overviewPath indexed, but when we try to retrieve the content doesn't actually exist?

**Comment by francinelucca** [comment]
this content is loaded runtime and requires remote mdx functionality so marking #321 epic as a blocker

**Comment by kingtraceyj** [comment]
Ideally, we would default to the usage page when there is no overview content. 

Then for the overview, instead of having just a blank page, it would be nice to have some sort of information about why there is a blank page. [Here's a good overview](https://blog.prototypr.io/a-ux-guide-for-designing-error-pages-fb9ced1f1c8a) of best practices or 404's or maintenance pages. I'm thinking something simple like: 
![image](https://user-images.githubusercontent.com/20582179/181587318-8787946b-b2b2-4c17-8337-07cd9b457c0b.png)

**Comment by mattrosno** [comment]
@kingtraceyj the dashboard will always be there, even if there are errors getting overview content.

**Comment by kingtraceyj** [comment]
ah. i see. as long as there is something there then that's ok!

**Comment by mattrosno** [comment]
@francinelucca this issue is no longer blocked by design response. If overviewPath is indexed, but it's invalid and the content doesn't actually exist, because the dashboard will always be there, @kingtraceyj is okay with doing nothing (not showing an error.)
